hrm-domain/lambdas/custom-integrations/uscr-beacon-poller/src/index.ts

```diff
@@ -14,46 +14,41 @@
 
-const accountSid = process.env.ACCOUNT_SID;
-const staticKey = process.env.STATIC_KEY;
-const lastUpdateSeenSsmKey = `/${process.env.NODE_ENV}/hrm/custom-integration/uscr/${accountSid}/latest_beacon_update_seen`;
+const lastIncidentReportUpdateSeenSsmKey = `/${process.env.NODE_ENV}/hrm/custom-integration/uscr/${accountSid}/beacon/latest_incident_report_seen`;
+const lastCaseReportUpdateSeenSsmKey = `/${process.env.NODE_ENV}/hrm/custom-integration/uscr/${accountSid}/beacon/latest_case_report_seen`;
 
-export const handler = async (): Promise<0> => {
-  const maxIncidents = parseInt(process.env.MAX_INCIDENT_REPORTS_PER_CALL || '1000');
-  // Read the last update seen from SSM
-  const lastUpdateSeen = await getSsmParameter(lastUpdateSeenSsmKey);
-  console.info('Last beacon update before:', lastUpdateSeen);
-  // Query Beacon API
-  const url = `${process.env.BEACON_URL}?updatedAfter=${lastUpdateSeen}&max=${maxIncidents}`;
-  console.info('Querying:', url);
-  const response = await fetch(url);
-  console.debug(
-    'External API responded with status:',
-    response.status,
-    await response.text(),
-  );
-  // Do something on the internal HRM API - will return a 404
-  const hrmResponse = await fetch(
-    `${process.env.INTERNAL_HRM_URL}/v0/accounts/${accountSid}/profiles/identifier/1234/flags`,
-    {
-      method: 'GET',
-      headers: {
-        Authorization: `Basic ${staticKey}`,
-      },
+export const handler = async ({
+  api,
+}: {
+  api: 'incidentReport' | 'caseReport';
+}): Promise<0> => {
+  const API_POLL_CONFIGS = {
+    caseReport: {
+      url: new URL(`${process.env.BEACON_BASE_URL}/caseReport`),
+      headers: beaconHeaders,
+      lastUpdateSeenSsmKey: lastCaseReportUpdateSeenSsmKey,
+      itemProcessor: addCaseReportSectionsToAseloCase,
+      maxItemsInChunk: parseInt(process.env.MAX_CASE_REPORTS_PER_CALL || '1000'),
+      maxChunksToRead: parseInt(process.env.MAX_CONSECUTIVE_API_CALLS || '10'),
+      itemTypeName: 'case report',
     },
-  );
-  console.debug(
-    'HRM API responded with status:',
-    hrmResponse.status,
-    await hrmResponse.text(),
-  );
-  // Update the last update seen in SSM
-  await putSsmParameter(lastUpdateSeenSsmKey, new Date().toISOString(), {
-    overwrite: true,
-  });
-  console.info('Last beacon update after:', await getSsmParameter(lastUpdateSeenSsmKey));
+    incidentReport: {
+      url: new URL(`${process.env.BEACON_BASE_URL}/incidentReport`),
+      headers: beaconHeaders,
+      lastUpdateSeenSsmKey: lastIncidentReportUpdateSeenSsmKey,
+      itemProcessor: addSectionToAseloCase('incidentReport', incidentReportToCaseSection),
+      maxItemsInChunk: parseInt(process.env.MAX_INCIDENT_REPORTS_PER_CALL || '1000'),
+      maxChunksToRead: parseInt(process.env.MAX_CONSECUTIVE_API_CALLS || '10'),
+      itemTypeName: 'incident report',
+    },
+  } as const;
 
+  await readApiInChunks<any>(API_POLL_CONFIGS[api]);
   return 0;
```

Comment:
Simply returning 0 in any case seems like poor error reporting, but there might be something I don't understand about how we do this.

Reply:
We report errors via thrown exceptions than return codes